cli/pcluster/update/update_command.py

```diff
@@ -206,3 +209,47 @@ def _restore_desired_size(cfn_params, stack_name, scheduler):
     # If there are compute nodes running, preserve current desired capacity (if possible)
     if int(cfn_params["MinSize"]) <= desired_capacity <= int(cfn_params["MaxSize"]):
         cfn_params["DesiredSize"] = str(desired_capacity)
+
+
+def _check_cluster_models(base_config, target_config, cluster_template):
+    """
+    Check if the cluster models of the two configurations are compatible with the update operation.
+
+    If target configuration is SIT and base is HIT a conversion is required.
+    """
+    same_cluster_model = target_config.cluster_model == base_config.cluster_model
+    if not same_cluster_model:
+        conversion_supported = (
+            True
+            if (target_config.cluster_model == ClusterModel.SIT and base_config.cluster_model == ClusterModel.HIT)
+            else False
+        )
+        if conversion_supported:
+            LOGGER.error(
+                (
+                    "The configuration of the cluster section in the '{config_file}' must be converted to the latest "
+                    "format with support for multiple queues before proceeding with the update.\n"
+                    "Please run the following command:\n"
+                    # TODO: update conversion command line when tool is available
+                    "pcluster-utils convert_config_file -c {config_file}{cluster_template_arg} -o {converted_file}\n"
+                    "Then retry with your converted configuration file by running the following command:\n"
+                    "pcluster update -c {converted_file} {cluster_name}"
+                ).format(
+                    config_file=target_config.config_file,
+                    converted_file="<new file>",
+                    cluster_name=base_config.cluster_name,
+                    cluster_template_arg=" -t " + cluster_template if cluster_template else "",
+                )
+            )
+        else:
+            LOGGER.error(
+                (
+                    "The configuration of the cluster section in the '{config_file}' configuration "
+                    "file seems incompatible with the CloudFormation Stack of the cluster '{cluster_name}'.\n"
```

Comment:
seems incompatible with the CloudFormation Stack --> I don't use "seems". I'd use "is" if it's the case.

CloudFormation Stack --> of the running cluster. We shouldn't talk about CloudFormation stack because the user cannot be aware of it.

Reply:
Fixed